subsys/net/ip/Kconfig.tls

```diff
@@ -0,0 +1,180 @@
+# Kconfig.stack - TLS/DTLS related options
+
+#
+# Copyright (c) 2018 Intel Corporation
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menu "TLS options"
+
+config NET_TLS
+	bool "TCP TLS support [EXPERIMENTAL]"
+	default n
+	depends on NET_TCP && MBEDTLS
+	help
+	  Enable TLS support for TCP connections. TLS is implemented by mbedTLS.
+
+config NET_DTLS
+	bool "UDP DTLS placeholder [EXPERIMENTAL]"
+	default n
+	depends on NET_UDP && MBEDTLS
+
+if NET_TLS || NET_DTLS
+
+menu "Supported TLS version"
+
+config TLS_VERSION_1_0
```

Comment:
Based on the comments above, I'd suggest to make this be a *truly* generic (mbed)TLS config for Zephyr, not tying it into NET_TLS || NET_DTLS (as indeed, there's nothing specific re: NET_TLS || NET_DTLS in selecting of TLS version).